### PR TITLE
201/docslinks

### PR DESCRIPTION
**Pull request by reyesrico**
This pull request creates a designed button for download Unknown Attachments sent from the bot to the user. Issue #201

### CODE REVIEW

File: src/Attachment.tsx [hunk 1]
```diff
@@ -310,6 +311,8 @@ export const AttachmentView = (props: {
 
         default:
             var unknownAttachment = props.attachment as UnknownMedia;
+            if (unknownAttachment.thumbnailUrl)
+                return <Unknown format={ props.format } contentType={ unknownAttachment.contentType } contentUrl={ unknownAttachment.contentUrl } name={ unknownAttachment.name } thumbnailUrl={ unknownAttachment.thumbnailUrl } />
```
Comment (billba) [inline src/Attachment.tsx]: Do you need this branch at all? Can't you just always return that first version, which will do the right thing if unknownAttachment.thumbnailUrl is undefined?

File: src/scss/botchat.scss [hunk 2]
```diff
@@ -700,4 +700,22 @@
...
+        width: 16px;        
+    }
+    
+    .wc-text-download {
+        display: inline-block;
+        text-decoration: none;
+        font-family: Segoe UI Semibold;
+        font-size: 15px;
+        color: #000000;
+    }
```
Comment (billba) [inline src/scss/botchat.scss]: Has this been reviewed with a designer?

File: src/scss/botchat.scss [hunk 3]
```diff
@@ -704,4 +704,51 @@
...
...4bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz4KPHN2ZyB3aWR0aD0iMTJweCIgaGVpZ2h0PSIyMHB4IiB2aWV3Qm94PSIwIDAgMTIgMjAiIHZlcnNpb249IjEuMSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayI+CiAgICA8IS0tIEdlbmVyYXRvcjogU2tldGNoIDQ1LjIgKDQzNTE0KSAtIGh0dHA6Ly93d3cuYm9oZW1pYW5jb2RpbmcuY29tL3NrZXRjaCAtLT4KICAgIDx0aXRsZT7uopYgY29weSA0PC90aXRsZT4KICAgIDxkZXNjPkNyZWF0ZWQgd2l0aCBTa2V0Y2guPC9kZXNjPgogICAgPGRlZnM+PC9kZWZzPgogICAgPGcgaWQ9IjIxNi0vLTMyMC0tLTAyMTYxNyIgc3Ryb2tlPSJub25lIiBzdHJva2Utd2lkdGg9IjEiIGZpbGw9Im5vbmUiIGZpbGwtcnVsZT0iZXZlbm9kZCI+CiAgICAgICAgPGcgaWQ9ImRvd25sb2FkX3dlYl9jaGF0IiB0cmFuc2Zvcm09InRyYW5zbGF0ZSgtMTgwLjAwMDAwMCwgLTEyNC4wMDAwMDApIiBmaWxsPSIjMkIyQjJCIj4KICAgICAgICAgICAgPHBhdGggZD0iTTE4MC40MzQ3ODMsMTQ0IEwxODAuNDM0NzgzLDE0Mi43NSBMMTkxLjU2NTIxNywxNDIuNzUgTDE5MS41NjUyMTcsMTQ0IEwxODAuNDM0NzgzLDE0NCBaIE0xOTIsMTM0LjQzOTQ1MyBMMTg2LDE0MC41NTI3MzQgTDE4MCwxMzQuNDM5NDUzIEwxODAuODY5NTY1LDEzMy41NjA1NDcgTDE4NS4zODE2NDMsMTM4LjE1MDM5MSBMMTg1LjM4MTY0MywxMjQgTDE4Ni42MTgzNTcsMTI0IEwxODYuNjE4MzU3LDEzOC4xNTAzOTEgTDE5MS4xMzA0MzUsMTMzLjU2MDU0NyBMMTkyLDEzNC40Mzk0NTMgWiIgaWQ9Iu6ili1jb3B5LTQiPjwvcGF0aD4KICAgICAgICA8L2c+CiAgICA8L2c+Cjwvc3ZnPg==');
```
Comment (danmarshall) [inline src/scss/botchat.scss]: Let's not base64 the SVG - can you leave it as plain xml?

File: src/Attachment.tsx [hunk 4]
```diff
@@ -123,12 +123,13 @@ const Unknown = (props: {
     format: FormatState,
     contentType: string,
     contentUrl: string,
-    name: string
+    name: string,
+    thumbnailUrl?: string
```
Comment (danmarshall) [inline src/Attachment.tsx]: is thumbnailUrl used?

File: src/scss/botchat.scss [hunk 5]
```diff
@@ -704,4 +704,51 @@
...
+    .wc-text-download {
+        display: inline-block;
+        font-family: Segoe UI Semibold;
+        font-size: 15px;        
+        text-decoration: none;
+        vertical-align: top;
+    }
+   
+    a.wc-link-download:link .wc-icon-download, a.wc-link-download:visited .wc-icon-download {
+        background-image:url('data:image/svg+xml;utf8,<svg width="12px" height="20px" viewBox="0 0 12 20" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><g id="216-/-320---021617" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd"><g id="download_web_chat" transform="translate(-180.000000, -124.000000)" fill="#2B2B2B"><path d="M180.434783,144 L180.434783,142.75 L191.565217,142.75 L191.565217,144 L180.434783,144 Z M192,134.439453 L186,140.552734 L180,134.439453 L180.869565,133.560547 L185.381643,138.150391 L185.381643,124 L186.618357,124 L186.618357,138.150391 L191.130435,133.560547 L192,134.439453 Z" id="c4"></path></g></g></svg>');
```
Comment (danmarshall) [inline src/scss/botchat.scss]: can you run this SVG through SVGO: https://jakearchibald.github.io/svgomg/

File: src/scss/botchat.scss [hunk 6]
```diff
@@ -704,4 +704,45 @@
...
+        display: inline-block;
+        height: 20px;
+        margin-left: 8px;
+        vertical-align: top;
+        width: 12px;        
+    }
+    
+    .wc-text-download {
+        display: inline-block;
+        font-family: Segoe UI Semibold;
```
Comment (danmarshall) [inline src/scss/botchat.scss]: We don't have this font. Maybe you can remove this.

**Pull request merged**